"Toolbox Arm" printable at exosuit fabricators and protolathes.

Mack3r3l opened the pull request:
I went ahead and made it possible for folk to print the toolset arm augmentation at exosuit fabricators. 

I also took the liberty to screw with the origin tech so it'd be _slightly_ more balanced.

Mack3r3l committed:
Added the ability to print the "ToolBox Arm" from exosuit fabricators.
(/obj/item/organ/internal/cyberimp/arm/toolset)

Signed-off-by: idontwannasignup  CKEY:Mack3r3l

Mack3r3l committed:
Decreased the "ToolBox Arm"'s orgin tech by 1 across the board.

Signed-off-by: idontwannasignup  CKEY:Mack3r3l

Mack3r3l committed:
Lowered origin tech by 1 in engineering and materials resulting in them both being 4.

Mack3r3l committed:
Merge branch 'master' of https://github.com/idontwannasignup/-tg-station

# Conflicts:
#	code/modules/surgery/organs/augments_arms.dm

Mack3r3l committed:
Origin tech for engineering and materials was lowered. Power manipulation went up by 1.

Signed-off-by: idontwannasignup CKEY: Mack3r3l

Mack3r3l committed:
dddddd

Mack3r3l committed:
Added the "Toolbox Arm" augmentation available for printing at exosuit fabricators.

Mack3r3l committed:
Added the "Toolbox Arm" augmentation available for printing at exosuit fabricators.

Signed-off-by: idontwannasignup CKEY: Mack3r3l

Mack3r3l committed:
Merge branch 'master' of https://github.com/idontwannasignup/-tg-station

Mack3r3l committed:
Added the "Toolbox Arm" augmentation to the exosuit fabricator MISC printing list.

Signed-off-by: idontwannasignup CKEY: Mack3r3l

idontwannasignup commented:
Also I made like previous commits because I'm new to this.

GunHog commented:
How do you use that thing again?

coiax commented:
Organ manipulation surgery targeted at the arm you're putting it in. Make sure it's configured to the correct arm type with a screwdriver.

kevinz000 commented:
This needs to be proto lathe too every other implant is like that

idontwannasignup commented:
I'd honestly love for it to be tested out in robotics first then see where I can take it from there since Core abandoned it.

PKPenguin321 commented:
i'm not sure the tool arm was ever actually finished
@Core0verload any thoughts on this?

jughu commented:
I thought it was finished just a tool upgrade with double speed on yoir arm that you have to switch by pressing the demn button unless it be cool if it autoselects the correct tool

idontwannasignup commented:
Core left me the tool arm to be implemented with his blessings so I imagine it's finished.
This was through IRC of course.

jughu commented:
Make it cost gold/silver so it isnt that common i mean like its double speed and doesnt take up anythinb

idontwannasignup commented:
If I have to add gold, I'm going to think about tweaking the origin tech and research requirements.

jughu commented:
I mean like even medhud implant requires some gold because it isnt meant to be mass produced and having tool arms mass produce is a bad idea since the tool arm welder has 80 fuel storage plus everything works on double speed

PKPenguin321 commented:
actually yeah now that I look at the values this is way too cheap, needs to unlock post-combat shotguns and should have a cost that requires some gold/silver

Core0verload commented:
It's finished, but it's UI still sucks.

kevinz000 commented:
gold/silver definately for balance reasons, combat shotgun level maybe-maybe, but no diamonds.
Or if that's absolutely nessacary for "balance", not too much. Maybe like 100/200 diamonds (One piece is 2000 materials I think) but not more because diamonds are rare enough already.

GunHog commented on code/modules/research/designs/mechfabricator_designs.dm:
```diff
@@ -688,3 +688,14 @@
 	build_path = /obj/item/drone_shell
 	category = list("Misc")
+
+/datum/design/toolarm
```
You should put this in https://github.com/tgstation/-tg-station/blob/master/code/modules/research/designs/medical_designs.dm with the others, and make it usable for both mech fab and protolathe. Any of the implants made on the protolathe are usually brought to robotics for implantation anyway.

idontwannasignup replied:
When you put it that way, I see the logic in it, good idea.

GunHog commented:
I would not ask for rare minerals for something this mid-level, to be honest. You are probably going to have access to a toolbelt, along with each tool if you can get access to this. Maybe gold and/or for the swag factor, but certainly not uranium or diamond. I say that only because of the emag option.

The fact that it requires surgery is enough to prevent it being something everyone will have. Look at welding shields - they are quite useful for their insignificant cost, but are not widely adopted due to needing surgery. Players are lazy.

Core0verload commented:
Hah. I remember there were a lot of balance retards in my welding shield PR (#11162), bitching about how imbalanced it is and how it must have drawbacks.

GunHog commented:
Then Lumi owes PKP $20, lol.

PKPenguin321 commented:
Reminds me of when people thought teleprods would be OP and now they're literally unused hahaha :sob:

Mack3r3l committed:
Added the "Toolbox Arm" cyberimp to medical_designs.dm

Removed the "Toolbox Arm" cyberimp from mechfabricator_designs.dm

Added "Protolathe" and "Mechfab" to the build_type in medical_designs.dm for the "Toolbox Arm"

Increased amount of metal it takes to produce.

Signed-off-by: idontwannasignup CKEY: Mack3r3l

GunHog commented on code/modules/research/designs/medical_designs.dm:
```diff
@@ -242,6 +242,17 @@
+	materials = list (MAT_METAL = 3500, MAT_GLASS = 500)
+	construction_time=200
+	build_path = /obj/item/organ/internal/cyberimp/arm/toolset
+	category = list("Misc")
```
`category = list("Misc", "Medical Designs")`

Mack3r3l committed:
Update medical_designs.dm

from category = list("Misc")

to category = list("Misc", "Medical Designs")

Good catch GunHog

idontwannasignup commented:
I can't think of anything else to add or remove before it's tested out by other players. 

Maybe I'll test the damage with the emagged knife in my own little world and see if it's not changling "armblade" powerful.

PKPenguin321 commented:
i still think it should at least cost some silver

Mack3r3l committed:
+Silver / +Biotech Tech Requirement

Adding a little bit of silver before you're able to build the augment.

Increased Biotech after looking at some of the other things such as HUD implants.

Mack3r3l committed:
+Power tech requirement/-build time/+materials

Reduced the time it takes to build, increased "powerstorage" by one in previous commit but forgot to include, changing it to 4. 
Went ahead and moved metal around so it uses only 1 sheet.

Shortened the item description as well.

idontwannasignup commented:
Went ahead and added silver into the mix considering it's an "uncommon" material. 

I did so for security because the ion rifle they have would just fuck everything up near the shot anyways.

jughu commented:
Altough you can still deconstruct a sugery tae for 2 silver but nobody would do that right? +1

KorPhaeron commented:
>costs 1/4 of an ore of silver

Very productive argument you guys had over that cost

KorPhaeron commented on code/modules/research/designs/medical_designs.dm:
```diff
@@ -242,6 +242,17 @@
+	id = "tool_arm"
+	req_tech = list("materials" = 4, "engineering" = 3, "biotech" = 4, "powerstorage" = 4)
+	build_type = PROTOLATHE | MECHFAB
+	materials = list (MAT_METAL = 2000, MAT_GLASS = 750, MAT_SILVER = 600)
```
Reminder a single sheet of material is 2k units.

Probably wont matter either way since people are drowning in every material but diamond and uranium, but being able to make ~12 of these on average from mining a single silver vein is silly. It's basically free.

idontwannasignup replied:
> Very productive argument you guys had over that cost

Kind of a dick way to say it's not enough silver but I get your point.

KorPhaeron replied:
Guess that was sorta rude, sorry

Mack3r3l committed:
Increased mat requirement and build time

Increased materials for metal out of consideration for how much it takes to make an actual toolbox vs the tools combined.

Glass boosted for consideration of the retractable emagged knife and welders.

And silver because by "uncommon" material, I mean I can boost it and we survive without having to worry about running out like we do with diamonds or uranium like Kor said earlier.

Signed off by ckey: Mack3r3l

idontwannasignup commented:
I really am looking for further comments regarding this, if not I'm assuming everyone who involves themselves with these sorts of things thinks it looks good.

GunHog commented:
Just do not jack up the price again! I think it is good as is.

PKPenguin321 commented:
We can always change the tech requirement and price later
I think this is mostly fair (although I hate giving science more shit but what can you do)

idontwannasignup commented:
> (although I hate giving science more shit but what can you do)

This started out for robotics considering no one bothers with it much since secborgs and ripleys are removed/useless

PKPenguin321 commented:
robotics *is* science (or really, science is robotics) fyi

KorPhaeron commented:
@idontwannasignup 

Merge conflicts

KorPhaeron commented:
Reopen this when you fix conflicts
